### ISSUE TITLE
GET /runtime/spec not in line with endpoint specs

### ISSUE THREAD

**Issue by CrommVardek** [paritytech/substrate-api-sidecar#624]
Hi, not a big issue:

**Description**

When requesting /runtime/spec (GET), I was expecting chainType to be one of the 3 values specified in the spec (live, development, local) as a string but instead I got an object "chainType":{"live":null} 

**Steps to Reproduce**

Using Docker image parity/substrate-api-sidecar:latest (v9.1.0)

GET on /runtime/spec

**Expected vs. Actual Behavior**

Expected property "chainType" of the response body to be a string, got an object instead.

Is it the swagger description ([endpoint description](https://paritytech.github.io/substrate-api-sidecar/dist/)) that is misleading or the response body is not correct ?

Thanks :)

**Comment by TarikGul** [comment]
Hey thanks for reporting this issue! Looking further into it, yes the issue is with the docs and the example itself, when looking at the schema it's correct in the sense that the ChainType is an `enum` that will either return:

```
{"development": null}
------------------------ or
{"live": null}
------------------------ or
{"local": null}
```

There are two different things that can get fixed here. 

1. the docs somehow need to be overridden, and the example updated. 

2. We change the `sanitizeNumbers` logic so that it abstracts the ChainType type to return a string. 

I will make sure to look into this further tomorrow, and propose a solution to get this straightened out. 

Thanks again for opening the issue.

**Comment by emostov** [comment]
> We change the sanitizeNumbers logic so that it abstracts the ChainType type to return a string.

I would recommend against this. The reason for this change looks to be that the enum ChainType got updated to include a variant that holds a value. If we try converting to a string I think we would lose some info for the `Custom(String)` variant

```
pub enum ChainType {
	/// A development chain that runs mainly on one node.
	Development,
	/// A local chain that runs locally on multiple nodes for testing purposes.
	Local,
	/// A live chain.
	Live,
	/// Some custom chain type.
	Custom(String),
}
```

**Comment by CrommVardek** [comment]
@TarikGul Thanks !

@emostov: Indeed, when I look at {"live":null} it seems to be holding a value. But this means that there are 4 possible chainTypes (not 3). However I find it a bit confusing to see an object that can have 4 differents "shapes" with no properties in common. 

If I may suggest something like "chainType":{"chainEnvironment":"live", "property":"AnyProps"} (where chainEnvironment value is "live", "development", "local" or "custom") ? It's just an idea to keep "chainType" consistent and easier to use but it might go against your design/schema...

Regardless, thanks for your quick responses :)

**Comment by TarikGul** [comment]
@emostov, Ahh yes I skimmed over it last night so I overlooked it but you are totally right. 

I think for now a good solution is to fix up the docs, and get the example up and accurate. 

When it comes to the return value for now, I think it would be wise to keep it as is, because it allows us to keep in line with `polkadot-js` and `substrate` as much as possible. For one we wouldn't have to introduce a breaking change to API, but more importantly it will be easier to spot and keep up to date with any changes that happens with `@polkadot/api` if they were to happen. 

> If I may suggest something like "chainType":{"chainEnvironment":"live", "property":"AnyProps"} (where chainEnvironment value is "live", "development", "local" or "custom") ? It's just an idea to keep "chainType" consistent and easier to use but it might go against your design/schema...

I do think this holds a lot of merit as the ex: `{"live": null}` isnt necessarily an ideal return value, but would just need some more discussion behind it because what I stated above, but for now I think it wise to not introduce any breaking changes, and have the return value be the same expected value that polkadot-js would return.
